### ISSUE TITLE
Aria label needed for copyable link container

### ISSUE THREAD

**Issue by LibbyUX** [hubmapconsortium/hra-ui#2122]
# Summary
While browsing our components, I found Storybook showing an accessibility violation on our Copy Link Container.

### Resources
- [Storybook: Copyable URL container](https://cdn.humanatlas.io/ui/storybook/design-system/?path=/story/design-system-copyable-url-container--primary)
- https://dequeuniversity.com/rules/axe/4.11/link-name?application=axeAPI
- https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA7.html

## Todo
- [ ] Fix aria label for this component

<img width="1479" height="1372" alt="Image" src="https://github.com/user-attachments/assets/3b6cf152-c019-4d67-ade6-27a1e4b685b0" />
